### DISCUSSION TITLE
Next.js 12 Supabase Auth Middleware

### DISCUSSION THREAD

**Discussion by ethndotsh** [supabase/supabase#5651] [Questions]
I am currently using Next.js 12's new `_middlware` feature and trying to use it to check if there is a user signed in on restricted routes (anything under `/app`). My current implementation is somewhat buggy because sometimes when you visit another page that is not restricted it will ask you to log in again when you return to a restricted page. Also, when you press the Back button in Chrome, it will sometimes break and ask you to log in again.

Here is my code: https://gist.github.com/cursecodes/d1c09f2c984485ddbbfccb3a16ad4810
And here are my Supabase Auth settings:
![supabase-auth-settings](https://user-images.githubusercontent.com/58047023/155587395-57d74c21-af4f-4be4-9a53-57f35a7e7a2e.png)

**Comment by Hallidayo** [comment]
I've been thinking of using Next.js middleware in my own project as I'm currently using a mix of a custom Context wrapper and getServerSideProps to check for authentication on my project a whole. 

When you say the following, do you get any errors to your console on the server?

> My current implementation is somewhat buggy because sometimes when you visit another page that is not restricted it will ask you to log in again when you return to a restricted page. Also, when you press the Back button in Chrome, it will sometimes break and ask you to log in again.

I found that the following [blog](https://jitsu.com/blog/supabase-nextjs-middleware) was useful and you could check this out - [source code](https://github.com/jitsucom/supabase-nextjs-middleware/blob/main/pages/index.tsx)

**Reply by ethndotsh** [reply]
That's the middleware I'm currently using. And no, there are no console errors. The user just shows up as null.
